pkg/ds/farm.go

```diff
@@ -262,8 +262,13 @@ func (dsf *Farm) handleDSChanges(changes dsstore.WatchedDaemonSets, quitCh <-cha
 		return
 	}
 
+	if changes.Total == 0 {
+		dsf.logger.Errorf("No daemon sets present, asked to delete %d; farm refuses to delete its %d daemon sets", len(changes.Deleted), len(dsf.children))
```

Comment:
hmmmm that's a good observation.

Maybe we should treat it as a soft-panic and abort the whole farm and start over?